server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -195,7 +201,7 @@ String innerDetect(ThreadMXBean threadBean, long currentThreadId) throws Excepti
             throw new ElasticsearchException("thread CPU time is not supported on this JDK");
         }
 
-        if (type == ReportType.MEM && SunThreadInfo.isThreadAllocatedMemorySupported() == false) {
+        if (type == ReportType.MEM && sunThreadInfo.isThreadAllocatedMemorySupported() == false) {
```

Comment:
Add an argument `SunThreadInfo sunThreadInfo` to innerDetect after ThreadMXBean to pass in the real sunThreadInfo from `detect()` or the mocked one during testing. This way we don't need to keep a reference of SunThreadInfo and expose a public or protected method on HotThreads.

Reply:
That's a good idea, I have  changed the code